webapp/src/shared/components/Input/SearchForm.tsx

```diff
@@ -0,0 +1,49 @@
+import { useEffect, useState } from 'react';
+import { Input, InputVariant } from '..';
+import { IconVariant } from '../Icon/Icon';
+import './SearchForm.css';
+
+type SearchProps<T> = {
+  url: string;
+  setChange: React.Dispatch<React.SetStateAction<T>>;
+};
+
+function useData<T>(url: string): T | null {
```

Comment:
can we put this somewhere in some folder like shared/hooks/ ? It's obvious that we will need it in other places

Reply:
Done in hooks/UseData.tsx